Topological_Sort/Topological_Sort.c

```diff
@@ -0,0 +1,84 @@
+#include <stdio.h>
+
+void topologicalsort(int a[][10],int num)
+{ 
+	int indegree[10],flag[10],count=0;
+	int k,j,i;
+
+	for(i=0;i<num;i++)
+	{
+		indegree[i]=0;
+		flag[i]=0;
+	}
+ 
+	for(i=0;i<num;i++)
+		for(j=0;j<num;j++)
+			indegree[i]=indegree[i]+a[j][i];
+ 
+	printf("\nThe topological order is:");
+ 
+	while(count<num)
+	{
+		for(k=0;k<num;k++)
+		{
+			if((indegree[k]==0) && (flag[k]==0))
+			{
+				printf("%d ",(k+1));
+				flag [k]=1;
+			}
+ 
+			for(i=0;i<num;i++)
+			{
+				if(a[i][k]==1)
+					indegree[k]--;
+			}
+		}
+ 
+		count++;
+	}
+}
+
+int main()
+{
+	int i,j,num,a[10][10];
+ 
+	printf("Enter the no of vertices:\n");
+	scanf("%d",&num);
+ 
+	printf("Enter the adjacency matrix:\n");
```

Comment:
Can you input the edges as a pair of vertices? A pair (u, v) denotes that there is an edge between u and v.

Reply:
@gupta2140 I tried to use pair (u,v) but for that vector are needed and the vector is not supported directly in C. 
Any suggestion?

Reply:
That's not an issue. You can input the number of vertices and the number of edges from the user. Then create an array of linked lists to simulate the adjacency list.

Reply:
@gupta2140 But in my opinion we cant create adjacency list with the number of vertices and the number of edges only. Because with the given number of vertices and edges, many different graphs can be made.

Reply:
Yes, that's why you need to input the number of vertices, edges and description of edges from the user. Thus the graph will be unique.
Suppose, n = 4 and m = 4 and edges are input by the user as:
1 2
2 3
3 4
4 1

Reply:
https://user-images.githubusercontent.com/42409030/55293730-4a5b7700-5417-11e9-8c98-48ed37786b17.png
@gupta2140 Please see the above link in that, I have made the different graph with the same number of vertices, edge, and description of edges.

Reply:
The graphs you made for those edge descriptions are wrong.
1. The graph should be directed.
2. Also, there should be no self-loops and cycles if you want to do topological sorting (in that case the vertices will go under a deadlock).

Here's the correct graph for both of your examples (corrected the cycles): https://pasteboard.co/I84Ah8T.jpg

Reply:
@gupta2140 I have done the required changes, please review.